greg-maxwell/2017-11-27-gmaxwell-advances-in-block-propagation.md

```diff
@@ -108,15 +108,15 @@ I created an omnibus protocol design where I took every good idea I could find,
lted, and the reason why they are salted is pretty straightforward. Even if we were to imagine that our short ids were 8 bytes long, so even less efficient, it would be very easy for a troublemaker to produce two transactions that have two short ids that are the same if it wasn't for the salt, and you could efficiently find two transactions with the same id and then you can spam the network with these and consequently jam up block propagation and a miner could do this in fact to create a profit potentially. So in the compact block scheme, this short id is salted using the block header that the block is found in. And when an attacker announces their transactions to the network, they aren't going to have known the salt because the block hasn't been found yet by definition. Also it has an additional salt that is provided by each of the generated parties that generate a compact block. The idea of this additional salt is that if just by dumb luck some transactions pair in the block or in the mempool happen to have the same short id as the one hashed with the block header-- by having different salts on different links, means that those collisions would happen sort of randomly on the network instead of happening everywhere and as a result blocks will propagate around slowdown. So if Alice and Bob connection gets slowdown because of a collision, the block would instead route through the fastest path. It starts off with the receiving peer requesting high bandwidth mode. Then in some arbitrary time in the future, a block shows up and a compact block is immediately send. You don't validate the block first, the validation box is moved down (referring to [this graph](https://raw.githubusercontent.com/bitcoin/bips/master/bip-0152/protocol-flow.png) on the slides), it occurs at the same time that you're sending out the block. The reason why this is okay is because by definition of the protocol it works this way; and high bandwidth mode only works with full validating nodes, so you'll notice if it's bad and you won't be handing it out to SPV nodes. Most of the time, some 85% of the time, that's where the protocol stops: the block shows up, the high bandwidth compact block peer sends it to you, and you're done. But if there was a hash collision or you didn't have a transaction, you have to disambiguate, and you ask with a getblocktxn message, and the remote side responds with the missing transactions you asked for. The low bandwidth mode works similarly, but you have to validate before participating. Just like the original block transmission, you ask "would you like the block" and they respond with "yes, please send me the compact block". It adds an extra roundtrip, but it's more bandwidth efficient. In high bandwidth mode, if you ask multiple peers for high bandwidth service, you might get multiple copies of the compact block, which on average is maybe 13 kilobytes and you might get multiple copies in that mode. It wastes a little bandwidth but it's much faster.
```

Comment:
Not sure why the exact change is not highlighted here. 
For easiest review, the change part is "by having different salts on different links, means that those collisions would happen sort of randomly on the network instead of happening everywhere and as a result blocks will propagate around slowdown"